### PR TITLE
chore: update to 3.9.0 final

### PR DESCRIPTION
**Pull request by henryiii**
Python 3.9.0 was released yesterday. Nuget was fast, though Manylinux finally updated 5 hours ago ~~(do we pin a docker image? If not, this should just work?)~~. Bumping macOS and Windows to 3.9.0. The other requirements should possibly be bumped before a release is made with 3.9.0 final, I think (@joerick?).

Edit: Oops, they are pinned, just found that. Will update.

### CODE REVIEW

**Comment by YannickJadoul** [comment]
@henryiii, there's a nice script to do that, `bin/update_dependencies.py`. Just comment out the `pip-compile` part, if you don't want to update the `constraints.txt` files.

**Comment by henryiii** [comment]
That's what I'm working on running (though I assume updating the constraints is fine, latests updates might be helpful for 3.9?), but I'm running into`No such file or directory: './env27/bin/pip-compile'` and investigating.

**Comment by YannickJadoul** [comment]
Those are virtualenvs that @joerick uses to update the package versions, I believe. I just comment out these lines when updating the images:

https://github.com/joerick/cibuildwheel/blob/630653b786790c3dbfbadb22eb2eaa9cd53d9e08/bin/update_dependencies.py#L13-L29

**Comment by YannickJadoul** [comment]
Also, the PyPy failures should be fixed once #430 is merged.

**Comment by henryiii** [comment]
I'll just comment out the constraints part for now, looks like I'd have to use pyenv and setup multiple Pythons to run that update; assuming one of you already has it (will do it eventually). Looks like the macOS tests are failing due to PyPy 7.3.1? Or am I misreading that?

**Comment by henryiii** [comment]
By the way, do we need to have a 37 constraints file now? If you run it with 3.8, then `importlib-metadata==1.7.0` goes away.

**Comment by YannickJadoul** [comment]
> assuming one of you already has it (will do it eventually).

I don't. I believe it's part of @joerick's steps when making a new release.

> Looks like the macOS tests are failing due to PyPy 7.3.1? Or am I misreading that?

Yup, related to #371, https://foss.heptapod.net/pypy/pypy/issues/3229 and https://foss.heptapod.net/pypy/pypy/merge_requests/726.

> By the way, do we need to have a 37 constraints file now? If you run it with 3.8, then `importlib-metadata==1.7.0` goes away. And `zipp`.

We probably do, then, yes :-(

**Comment by henryiii** [comment]
@joerick Would you like me to add a env37/constraints37 setup, and start assuming that the "main" one runs 3.8 (which should be identical to 3.9, so a "38" constraints is still not needed)?

@YannickJadoul What do we do about PyPy? And why is it just now breaking?

**Comment by YannickJadoul** [comment]
> env37/constraints37 setup

Maybe you can add this to the list of versions in `bin/update_dependencies.py`, such that @joerick will get a (not-so-gentle) reminder, next time these dependencies are updated?

> @YannickJadoul What do we do about PyPy? And why is it just now breaking?

That I don't really know. The issue was linked to versions of the libraries installed in the environment. Did Azure update their images? (Always blame CI updates, when possible ;-) )

**Review by YannickJadoul — APPROVED** [review]
After merging #430, things should turn green. At any rate, the failures are not related to this PR.

**Comment by henryiii** [comment]
@joerick be sure to check 501ac92, I've added a 3.7 constraints file there and updated the docs on setting up a environment to run the update step (and, obviously, I ran the update step).

**Review by YannickJadoul — APPROVED** [review]
Hurray! :-)

**Comment by joerick** [comment]
Niiiiice work @henryiii ! Thanks for doing that, and updating the docs :) Yep, this looks good to me, I'll try to push a release soon so we have a Python 3.9.0 version out there.

**Review by joerick — APPROVED** [review]

**Comment by henryiii** [comment]
Any idea why one AppVeyor job basically timed out? https://ci.appveyor.com/project/joerick/cibuildwheel/builds/35599358/job/hiq0jff3cca76c4f 

Can someone restart it?

File: cibuildwheel/resources/pinned_docker_images.cfg [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+manylinux1 = quay.io/pypa/manylinux1_i686:2020-10-06-c4a4a7c
+manylinux2010 = quay.io/pypa/manylinux2010_i686:2020-10-06-786a931
+manylinux2014 = quay.io/pypa/manylinux2014_i686:2020-10-06-b2ca7a1
 
 [pypy_x86_64]
 manylinux2010 = pypywheels/manylinux2010-pypy_x86_64:2020-09-26-8f210b5
 
 [aarch64]
-manylinux2014 = quay.io/pypa/manylinux2014_aarch64:2020-09-26-7f7a14f
+manylinux2014 = quay.io/pypa/manylinux2014_aarch64:2020-10-05-2f3b046
```
Comment (mayeut) [inline cibuildwheel/resources/pinned_docker_images.cfg:15]: The aarch64 build failed on travis-ci fi-or the manylinux2014 aarch64 image. I restarted it and it should now be available.
```suggestion
manylinux2014 = quay.io/pypa/manylinux2014_aarch64:2020-10-06-b2ca7a1
```

Comment (henryiii) [reply cibuildwheel/resources/pinned_docker_images.cfg:15]: Good catch, didn't notice that! Reran `./bin/update_dependencies.py` and indeed, this is the only change.

**Comment by henryiii** [comment]
Passed and several approvals, so merged! :)

**Pull request merged**